qa/zcash/smoke_tests.py

```diff
@@ -0,0 +1,800 @@
+    # Mining
+    ('7a', True, True), # Run getblocksubsidy
+    ('7b', True, True), # Run getblocktemplate
+    ('7c', True, True), # Run getmininginfo
+    ('7d', True, True), # Run getnetworkhashps
+    ('7e', True, True), # Run getnetworksolps
+]
+
+
+#
+# Test helpers
+#
+
+def run_cmd(results, case, zcash, name, args=[]):
+    print('-----')
+    print('%s $ zcash-cli %s %s' % (
+        case.ljust(3),
+        name,
+        ' '.join([str(arg) for arg in args],
+    )))
+    try:
+        res = zcash.__getattr__(name)(*args)
+        print(res)
+        print()
+        if results is not None and len(case) > 0:
+            results[case] = True
+        return res
+    except RpcException as e:
+        print('ERROR: %s' % str(e))
+        if results is not None and len(case) > 0:
+            results[case] = False
+        return None
+
+def wait_for_balance(zcash, zaddr, expected=None, timeout=900):
+    print('Waiting for funds to %s...' % zaddr)
+    if expected is not None:
+        print('Expecting %s' % expected)
+
+    unconfirmed_balance = zcash.z_getbalance(zaddr, 0)
+    print('Unconfirmed balance: %s' % unconfirmed_balance)
+    if expected is not None and unconfirmed_balance != expected:
+        print('WARNING: Unconfirmed balance does not match expected balance')
+
+    ttl = timeout
+    while True:
+        balance = zcash.z_getbalance(zaddr)
+        if balance == unconfirmed_balance:
+            print('Received %s' % balance)
+            return balance
+
+        time.sleep(1)
+        ttl -= 1
+        if ttl == 0:
+            # Ask user if they want to keep waiting
+            print()
+            ret = input('Do you wish to continue waiting? (Y/n) ')
+            if ret.to_lower() == 'n':
+                print('Address contained %s at timeout' % balance)
+                return balance
+            else:
+                ttl = 300
+
+def wait_and_check_balance(results, case, zcash, addr, expected):
+    balance = wait_for_balance(zcash, addr, expected)
+    if balance != expected and results is not None and len(case) > 0:
+        results[case] = False
+    return balance
+
+def wait_for_txid_operation(zcash, opid, timeout=300):
+    print('Waiting for async operation %s' % opid)
+    result = None
+    for _ in iter(range(timeout)):
+        results = zcash.z_getoperationresult([opid])
+        if len(results) > 0:
+            result = results[0]
+            break
+        time.sleep(1)
+
+    status = result['status']
+    if status == 'failed':
+        print('Operation failed')
+        print(result['error']['message'])
+        return None
+    elif status == 'success':
+        txid = result['result']['txid']
+        print(txid)
+        return txid
+
+def async_txid_cmd(results, case, zcash, name, args=[]):
+    opid = run_cmd(results, case, zcash, name, args)
+    # Some async commands return a dictionary containing the opid
+    if type(opid) == type({}):
```

Comment:
I think this is fine here, but still worth mentioning: `do not compare types, use 'isinstance()'flake8(E721)`